awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -275,9 +283,18 @@ def update_fields(p):
         v = params.pop(old_k)
         params_update[new_k] = v
 
-    extra_vars = params.get('extra_vars_path')
-    if extra_vars is not None:
-        params_update['extra_vars'] = ['@' + extra_vars]
+    extra_vars = params.get('extra_vars')
+    extra_vars_path = params.get('extra_vars_path')
+
+    if extra_vars:
+        params_update['extra_vars'] = [json.dumps(extra_vars)]
+
+    elif extra_vars_path is not None:
+        params_update['extra_vars'] = ['@' + extra_vars_path]
+        module.deprecate(
+            msg='extra_vars_path should not be used anymore. Use \'extra_vars: "{{ lookup(\'file\', \'/path/to/file\') | from_yaml }}"\' instead',
```

Comment:
Oh you know what, it only now occurred to me, we should test that this works in a playbook. I could see the dict type breaking this.

Reply:
tested in https://github.com/ansible/test-playbooks/pull/99